src/instructlab/sdg/utilblocks.py

```diff
@@ -95,3 +95,82 @@ def generate(self, samples: Dataset) -> Dataset:
             self.separator,
             self.ctx.dataset_num_procs,
         )
+
+
+class FlattenColumnsBlock(Block):
+    """Melt/transform a data from a wide format to a long format see pandas.melt for a description
+
+    Args:
+            var_cols (list): Column(s) to unpivot. All other columns are set to use as identifier variables.
+            value_name (str): Name to use for the ‘value’ column, can’t be an existing column label.
+            var_name (str):  Name to use for the ‘variable’ column. If None it uses frame.columns.name or ‘variable’.
+    """
+
+    def __init__(
+        self, ctx, pipe, block_name: str, var_cols: list, value_name: str, var_name: str
+    ) -> None:
+        super().__init__(ctx, pipe, block_name)
+        self.var_cols = var_cols
+        self.value_name = value_name
+        self.var_name = var_name
+
+    def generate(self, samples: Dataset) -> Dataset:
+        df = samples.to_pandas()
+        id_cols = [col for col in samples.column_names if col not in self.var_cols]
+        flatten_df = df.melt(
+            id_vars=id_cols,
+            value_vars=self.var_cols,
+            value_name=self.value_name,
+            var_name=self.var_name,
+        )
+        return Dataset.from_pandas(flatten_df)
```

Comment:
This likely needs a `reset_index()` call - but let's resolve that with a new utility function: https://github.com/instructlab/sdg/issues/190